addon/mixins/frost-list-sorting-mixin.js

```diff
@@ -1,37 +1,32 @@
 import Ember from 'ember'
 const {
   Mixin,
-  on
+  get,
+  on,
+  set,
+  defineProperty,
+  computed: {alias}
 } = Ember
-import computed from 'ember-computed-decorators'
+import {normalizeSort, defaultSort} from 'ember-frost-list/utils/utils'
 import FrostListCoreMixin from 'ember-frost-list/mixins/frost-list-core-mixin'
 
 export default Mixin.create(FrostListCoreMixin, {
   // == Event =================================================================
+  // TODO replace defineProperty when there's a public method available
   initListSortingMixin: on('init', function () {
-    this.set('queryParams', ['listConfig.sorting.active'])
-    Ember.defineProperty(this, 'sortableProperties', Ember.computed.alias('listConfig.sorting.properties'))
-    Ember.defineProperty(this, 'activeSorting', Ember.computed.alias('listConfig.sorting.active'))
+    defineProperty(this, 'sortableProperties', alias('listConfig.sorting.properties'))
+    defineProperty(this, 'activeSorting', alias('listConfig.sorting.active'))
   }),
 
-  // == Computed Properties ====================================================
-  sortedItems: Ember.computed.sort('statefulListItems', 'activeSortingString'),
-
-  @computed('activeSorting')
-  activeSortingString (activeSorting) {
-    if (!activeSorting) return []
-    return activeSorting.map((sortProperty) => {
-      return `record.${sortProperty.value}${sortProperty.direction}`
-    })
-  },
-
   // == Actions ================================================================
   actions: {
-    sortItems (sortItems) {
-      let activeSorting = sortItems.map(function (item) {
-        return {value: item.value, direction: item.direction}
-      })
-      this.set('activeSorting', activeSorting)
+    sortItems (sortProperties) {
+      const normalizedSortProperties = normalizeSort(sortProperties)
+      const customSortMethod = get(this, 'listConfig.sorting.client')
+      const sortMethod = typeof customSortMethod === 'function' ? customSortMethod : defaultSort
```

Comment:
Are we using prop-types for the `sorting.client` type?  If so, we don't need this check (if not, we probably should use prop-types)

Basically I'm looking for an assertion instead of a silent fallback